### PR TITLE
VDS-2943, VDS-2945: Send invite to email

### PR DESCRIPTION
**Pull request by asvishnyakov**

### CODE REVIEW

File: src/XProfile/VirtoCommerce.XProfile/Authorization/ProfileAuthorizationHandler.cs [hunk 1]
```diff
@@ -1,213 +1,222 @@
...
+            }
+            else if (context.Resource is UpdatePersonalDataCommand updatePersonalDataCommand)
+            {
+                updatePersonalDataCommand.UserId = currentUserId;
+                result = true;
+            }
+            else if (context.Resource is InviteUserCommand inviteUserCommand && currentContact != null)
+            {
+                var currentUser = await _userManager.FindByIdAsync(currentUserId);
+                result = currentContact.Organizations.Contains(inviteUserCommand.OrganizationId) && currentUser?.StoreId == inviteUserCommand.StoreId;
```
Comment (pushnitsa) [inline src/XProfile/VirtoCommerce.XProfile/Authorization/ProfileAuthorizationHandler.cs:168]:
```suggestion
result = currentContact.Organizations.Contains(inviteUserCommand.OrganizationId) && currentUser?.StoreId.EqualsInvariant(inviteUserCommand.StoreId);
```

File: src/XProfile/VirtoCommerce.XProfile/Schemas/InputRegisterByInvitationType.cs [hunk 2]
```diff
@@ -0,0 +1,18 @@
+using GraphQL.Types;
+using VirtoCommerce.ExperienceApiModule.XProfile.Commands;
+
+namespace VirtoCommerce.ExperienceApiModule.XProfile.Schemas
+{
+    public class InputRegisterByInvitationType: InputObjectGraphType
+    {
+        public InputRegisterByInvitationType()
+        {
+            Field<NonNullGraphType<StringGraphType>>(nameof(RegisterByInvitationCommand.UserId));
```
Comment (pushnitsa) [inline src/XProfile/VirtoCommerce.XProfile/Schemas/InputRegisterByInvitationType.cs:10]: Add description please

File: src/XProfile/VirtoCommerce.XProfile/Schemas/InputInviteUserType.cs [hunk 3]
```diff
@@ -0,0 +1,17 @@
+using GraphQL.Types;
+using VirtoCommerce.ExperienceApiModule.XProfile.Commands;
+
+namespace VirtoCommerce.ExperienceApiModule.XProfile.Schemas
+{
+    public class InputInviteUserType : InputObjectGraphType
+    {
+        public InputInviteUserType()
+        {
+            Field<NonNullGraphType<StringGraphType>>(nameof(InviteUserCommand.StoreId));
```
Comment (pushnitsa) [inline src/XProfile/VirtoCommerce.XProfile/Schemas/InputInviteUserType.cs:10]: Add description please

File: src/XProfile/VirtoCommerce.XProfile/Authorization/ProfileAuthorizationHandler.cs [hunk 4]
```diff
@@ -1,213 +1,223 @@
...
+            }
+            else if (context.Resource is RegisterByInvitationCommand)
```
[the remark this answers is not in the document]
Comment (asvishnyakov) [reply src/XProfile/VirtoCommerce.XProfile/Authorization/ProfileAuthorizationHandler.cs]: Removed

File: src/XProfile/VirtoCommerce.XProfile/Schemas/InputInviteUserType.cs [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+
+namespace VirtoCommerce.ExperienceApiModule.XProfile.Schemas
+{
+    public class InputInviteUserType : InputObjectGraphType
+    {
+        public InputInviteUserType()
+        {
+            Field<NonNullGraphType<StringGraphType>>(nameof(InviteUserCommand.StoreId));
+            Field<NonNullGraphType<StringGraphType>>(nameof(InviteUserCommand.OrganizationId));
+            Field<NonNullGraphType<StringGraphType>>(nameof(InviteUserCommand.UrlSuffix));
```
[the remark this answers is not in the document]
Comment (asvishnyakov) [reply src/XProfile/VirtoCommerce.XProfile/Schemas/InputInviteUserType.cs]: Made optional

File: src/XProfile/VirtoCommerce.XProfile/Schemas/InputRegisterByInvitationType.cs (hunk 2 above)
Comment (asvishnyakov) [reply src/XProfile/VirtoCommerce.XProfile/Schemas/InputRegisterByInvitationType.cs]: Added

File: src/XProfile/VirtoCommerce.XProfile/Schemas/InputInviteUserType.cs (hunk 3 above)
Comment (asvishnyakov) [reply src/XProfile/VirtoCommerce.XProfile/Schemas/InputInviteUserType.cs]: Added

**Review by trueboroda — APPROVED** [review]

**Review by pushnitsa — APPROVED** [review]

**Review by Vectorfield4 — APPROVED** [review]

**Pull request merged**